package/MDAnalysis/topology/GSDParser.py

```diff
@@ -0,0 +1,138 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding: utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
+#
+# MDAnalysis --- https://www.mdanalysis.org
+# Copyright (c) 2006-2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+
+"""
+GSD topology parser
+=========================
+
+.. versionadded:: 0.17.0
+
+The :class:`GSDParser` generates a topology from files for the HOOMD_ code.
+
+TODO: write info
+
+.. _HOOMD: http://codeblue.umich.edu/hoomd-blue/index.html
+.. _HOOMD XML: http://codeblue.umich.edu/hoomd-blue/doc/page_xml_file_format.html
+
+Classes
+-------
+
+.. autoclass:: GSDParser
+   :members:
+   :inherited-members:
+
+"""
+from __future__ import absolute_import
+
+try :
+    import gsd.hoomd
+except :
+    pass
+import numpy as np
+
+from . import guessers
+from ..lib.util import openany
+from .base import TopologyReaderBase
+from ..core.topology import Topology
+from ..core.topologyattrs import (
+    Atomtypes,
+    Atomids,
+    Angles,
+    Bonds,
+    Charges,
+    Dihedrals,
+    Impropers,
+    Masses,
+    Radii,
+    Resids,
+    Resnums,
+    Segids,
+)
+
+
+class GSDParser(TopologyReaderBase):
```

Comment:
this class needs tests (separate to the reader tests).  check things like atomtypes, masses etc are all being read correctly.

Reply:
you mean that these tests should be included in the testsuite, or inside the class?